js/modules/bookList.js

```diff
@@ -0,0 +1,80 @@
+export default class BookListing {
+  // declare a constructor function
+  constructor() {
+    if (localStorage.getItem('bookList')) {
+      this.list = JSON.parse(localStorage.getItem('bookList'));
+    } else {
+      this.list = [];
+    }
+  }
+
+  // create a display method
+  displayList() {
+    // clear the current list UI
+    const bookListDiv = document.querySelector('.books');
+    bookListDiv.innerHTML = '';
+
+    // checking the number of books in the book list container
+    if (this.list.length === 0) {
+      // create a empty message element with a class 'book__empty'
+      const emptyMsg = document.createElement('p');
+      emptyMsg.className = 'book__empty';
+      emptyMsg.innerText = 'Empty Book Collection';
+
+      // Append the book card to the parent node
+      bookListDiv.appendChild(emptyMsg);
+    } else {
+      // Loop through the array given
+      this.list.forEach((book, id) => {
+        // create a book card with a class 'book'
+        const bookDiv = document.createElement('div');
+        bookDiv.className = 'book';
+
+        // create a book details p with the class 'book__details'
+        const bookDetails = document.createElement('p');
+        bookDetails.className = 'book__details';
+        bookDetails.innerText = `"${book.title}" by ${book.author}`;
+
+        // Create the delete button with a class 'book__remove-btn'
+        const deleteBtn = document.createElement('button');
+        deleteBtn.className = 'book__remove-btn';
+        deleteBtn.innerText = 'Remove';
+
+        // Attach a listener to the remove botton for delete function
+        deleteBtn.addEventListener('click', () => {
+          this.removeBookFromList(id);
+        });
+
+        bookDiv.appendChild(bookDetails);
+        bookDiv.appendChild(deleteBtn);
+
+        // Append the book card to the parent node
+        bookListDiv.appendChild(bookDiv);
+      });
+    }
+  }
+
+  // create a book method
+  addToList(newBook) {
+    // update the book list with the new book
+    this.list.push(newBook);
+
+    // update the local storage with the new books
+    localStorage.setItem('bookList', JSON.stringify(this.list));
+
+    // update the page with the new book
+    this.displayList();
+  }
+
+  // create delete book method
+  removeBookFromList(index) {
```

Comment:
- [ ] Great working on the modules and the design but I highly recommend making the functions to be arrow fucntions to follow ES6 best practices and once it is one of the requirements so kindly update it 💪🏻😊🔥🚀

Reply:
@Reem-lab Thanks for the feedback. However, those are methods in the BookList class. According to the resources given, below is a specimen of a typical ES6 class with a method. My approach is consistence with the given [reference](http://es6-features.org/#ClassDefinition) 
![reference](https://user-images.githubusercontent.com/22054464/200671755-074238d4-d6a9-4cd2-97c0-296d8fd6a146.png)

Reply:
[ Required ] As requested by a previous reviewer please make sure that the following methods in class `BookListing` are written using  `ES6` arrow functions syntax. The two methods are;
- displayList()
- addToList()
- removeBookFromList(index)

You can achieve this by using the code below

```JavaScript
  class BookListing {
     displayList = () => {
       // Body of the method
     }

    addToList = (newBook) => {
      // Body of the method
     }
     
     removeBookFromList = (index) => {
        // Body of the method
    }
  }
```